src/zkevm_specs/evm/execution/signextend.py

```diff
@@ -0,0 +1,54 @@
+from ..instruction import Instruction, Transition
+from zkevm_specs.util import FQ
+
+
+def signextend(instruction: Instruction):
+    opcode = instruction.opcode_lookup(True)
+
+    index = instruction.stack_pop()
+    value = instruction.stack_pop()
+    result = instruction.stack_push()
+
+    # Any index value >= 256 always returns the same value
+    is_msb_sum_zero = instruction.is_zero(FQ(sum(index.le_bytes[1:32])))
+    sign_byte = (
+        FQ((value.le_bytes[index.le_bytes[0]] >> 7) * 0xFF) if index.le_bytes[0] < 31 else FQ(0)
+    )
+    selectors = [FQ(index.expr() == FQ(i)) for i in range(31)]
```

Comment:
I found selectors logic seems a bit different from the circuit part. In circuit, seems it will be conservative 1, like `[0, 0, 0, 1 (index), 1, 1, 1]` but here looks like onehot `[0, 0, 0, 1, 0, 0, 0]`

Reply:
Thank you for the correct!
I fixed.
I would appreciate it if you could confirm.